socorro/siglists/__init__.py

```diff
@@ -0,0 +1,54 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+
+
+_DIRECTORY = os.path.dirname(os.path.abspath(__file__))
+
+
+# This is a hack because sentinels can be a tuple, with the second item being
+# a function to verify if the sentinel applies. It's quite hard to express
+# that in a .txt file, so this special value is here. This list should not
+# grow much, and if it does, we should find a better solution for handling
+# these special values.
+_SPECIAL_EXTENDED_VALUES = {
+    'signature_sentinels': (
+        (
+            'mozilla::ipc::RPCChannel::Call(IPC::Message*, IPC::Message*)',
+            lambda x: (
+                'CrashReporter::CreatePairedMinidumps(void*, '
+                'unsigned long, nsAString_internal*, nsILocalFile**, '
+                'nsILocalFile**)'
+            ) in x
+        ),
+    ),
+}
+
+
+def _get_file_content(source):
+    """Return a tuple, each value being a line of the source file.
+
+    Remove empty lines and comments (lines starting with a '#').
+    """
+    filepath = os.path.join(_DIRECTORY, source + '.txt')
+
+    with open(filepath, 'r') as f:
+        lines = tuple(
+            line.replace('\n', '') for line in f
```

Comment:
I'm not sure what this line does. Are you expecting the string "\n" to appear in the text? If so, then you need to do:

```
line.strip().replace('\\n', '')
```

If you're just removing the newline at the end, then you could do:

```
line.strip()
```

which removes all whitespace characters.

I didn't see any whitespace at the beginning and end of any of the lines. Is maintaining non-newline whitespace important?